src/NetMQ/NetMQPoller.cs

```diff
@@ -280,7 +297,7 @@ public void Remove([NotNull] NetMQTimer timer)
 
             timer.When = -1;
 
-            Run(() => m_timers.Remove(timer));
+            Run(() => m_timers.Remove(timer)).Wait();
```

Comment:
Again, synchronous waits are not safe. This would have to be a `RemoveAsync` overload.